cmake/modules/extensions.cmake

```
@@ -718,6 +718,222 @@ function(generate_inc_file_for_target
+#   OUTPUT        - The generated file.
+#   BINARY_FILES  - List of binary files to be converted into embeddable form.
+#   SYMBOL_NAMES  - List of symbol names corresponding to binary files being embedded.
+#
+# Optional named parameters:
+#
+#   OFFSET        - Offset, in bytes, from which data inclusion from the
+#                   specified files should be started.
+#   LENGTH        - Length, in bytes, of the data subset to be included from
+#                   the specified files.
+#   READONLY      - yes (default) or no.
+#   ALIGN         - Alignment for data being embedded in bytes. Default is 4.
+#   SECTION       - Name of the desired linker section. Data from all specified
+#                   binary files will be placed in the given section name.
+#   COMPRESS      - Compress the data with specified compression method: gzip or no (default).
+#   GZIP_MTIME    - mtime seconds in the gzip header.
+#                   Defaults to zero to keep builds deterministic. For
+#                   current date and time, set this option to "now".
+#
+# Examples:
+#
+#   generate_binary_embed_inc_file_for_target(app
+#     OUTPUT        ${gen_dir}/test_data_embed_combined.inc
+#     BINARY_FILES  test1.bin test2.bin
+#     SYMBOL_NAMES  symbol_names test1_data test2_data
+#     ALIGN         16
+#   )
+#
+#   generate_binary_embed_inc_file(
+#     OUTPUT        ${gen_dir}/test_data_embed.inc
+#     BINARY_FILES  test_data.bin
+#     SYMBOL_NAMES  test_data
+#     OFFSET        5
+#     LENGTH        52
+#     SECTION       tdata
+#     READONLY      no
+#     ALIGN         32
+#   )
+#
+#   generate_binary_embed_inc_file(
+#     OUTPUT        ${gen_dir}/raw_data.inc
+#     BINARY_FILES  ${EXTERNAL_SDK_PATH}/testx_file.bin
+#     SYMBOL_NAMES  test_data
+#   )
+#
+function(generate_binary_embed_inc_file_for_target target)
+  set(single_args "OUTPUT")
+  set(opt_single_args "SECTION;ALIGN;READONLY;OFFSET;LENGTH;COMPRESS;GZIP_MTIME")
+  set(multi_args "BINARY_FILES;SYMBOL_NAMES")
+  cmake_parse_arguments(PARSE_ARGV 1 BINARY_EMBED "${options}"
+    "${single_args};${opt_single_args}" "${multi_args}")
+
+  if(BINARY_EMBED_UNPARSED_ARGUMENTS)
+    message(FATAL_ERROR "generate_binary_embed_inc_file_for_target(${ARGV0} ...) given unknown "
+                        "arguments: ${BINARY_EMBED_UNPARSED_ARGUMENTS}"
+    )
+  endif()
+
+  foreach(arg ${single_args};${multi_args})
+    if(NOT DEFINED BINARY_EMBED_${arg})
+      message(FATAL_ERROR "generate_binary_embed_inc_file_for_target(${ARGV0} ...) "
+                          "missing required argument: ${arg}"
+      )
+    endif()
+  endforeach()
+
+  list(LENGTH BINARY_EMBED_BINARY_FILES bin_files_count)
+  list(LENGTH BINARY_EMBED_SYMBOL_NAMES symbol_names_count)
+  if(NOT symbol_names_count EQUAL bin_files_count)
+      message(FATAL_ERROR "Number of specified binary files (${bin_files_count}) is not the same "
+              "as number of the corresponding symbols names (${symbol_names_count})")
```

Please try to follow same error text pattern as used by the `zephyr_check_arguments_x` macros.

Something like
```suggestion
    message(FATAL_ERROR "${CMAKE_CURRENT_FUNCTION}(BINARY_FILES ${BINARY_EMBED_BINARY_FILES} ...) mismatch length of: SYMBOL_NAMES ${BINARY_EMBED_SYMBOL_NAMES}."
```